### ISSUE TITLE
Undo/redo issue

### ISSUE THREAD

**Issue by lytex** [nobiot/org-transclusion#60]
I've been using org transclusion for a couple of months and it's pretty good!

The most recurring issue I've had is that using doom emacs and evil mode, undo and redo do something strange and substitute without notice the transcluded element with its contents, effectively copying it

If I have a file main.org with transclusion and I undo and redo several times, I got from this:
```
#+transclude: t
[[file:whatever.org]]
#+transclude: t
[[file:ohno.org]]
```
to this:

```
* whatever
** a test headline
** these are the contents of the file whatever.org
*** these shouldn't be here
* ohno
** this is not meant to be here
** these are the contents of the file ohno.org
```

A simple workaround is to add the following lines to the config to disable transclusion when undoing/redoing:

```elisp
(advice-add 'evil-undo :before (lambda (count) (if org-transclusion-mode (org-transclusion-deactivate))))
(advice-add 'evil-redo :before (lambda (count) (if org-transclusion-mode (org-transclusion-deactivate))))
```

**Comment by nobiot** [comment]
@lytex thank you for this... Not sure what I can do on my end at the moment. Undo recognises text copy and overlay as separate steps. Maybe I should try grouping them as one unit of operation for undo. But not sure if that will resolve the issue of evil-undo/redo...

**Comment by lytex** [comment]
I'm trying different things, as it seems the advice is not working in all cases.
evil-undo calls evil-undo-function calls undo-fu-only-undo/undo-fu-only-redo, which calls undo, so I'm now advising undo and using plain undo instead of evil-undo.

In the video, I activate org-transclusion-mode and then call plain undo, a couple times. Advice is working ok
Then, I wrote something and undo now copies the transcluded tree.

https://user-images.githubusercontent.com/32138262/111546117-6fb62c80-8777-11eb-8e23-6c5b360c1694.mp4

Doom advises undo with `doom-modeline-update-buffer-file-name`, but it is advised after, so that org-transclusion should be disabled before that.

**Comment by nobiot** [comment]
Interesting... Thanks for your notes. I will definitely come back to this issue to see if I could do something...

**Comment by lytex** [comment]
Maybe a better temporary solution is to enable read mode when transclusion is on:

```lisp
(defun my/org-transclusion-read-only (&optional arg)
(if org-transclusion-mode
  (setq buffer-read-only t)
  (setq buffer-read-only nil)))

(advice-add 'org-transclusion-mode :after 'my/org-transclusion-read-only)
```

**Comment by nobiot** [comment]
Working on a major, breaking change (summarised in PR #69) on a new branch v0.1.0.
This one has a better undo/redo behavour as it is based on text-properties, not overview.

Not stable, but workable, if you would like to have a peek :)

**Comment by lytex** [comment]
Great!
Regarding the temporary solution, it has been working fine for me, and I haven't got any transclusion weirdness so far. 👍

I'll let you know when I try it!

**Comment by nobiot** [comment]
Merged v0.1.0 in the main branch. Undo/redo should be much more robust and intuitive. I have also added back the live-sync edit feature implemented in a new approach. It would be great if you could try some of the new features. You can find more in the new README. 

Closing for now. You can reopen this issue, or create a new one.

Thank you.

**Comment by lytex** [comment]
I cannot test it on doom emacs, even if it is freshly installed (see #71)
